hub.go

```diff
@@ -45,14 +47,47 @@ type hub struct {
 
 	// Unregister requests from connections.
 	unregister chan *connection
+
+	// Serial hub to communicate with serial ports
+	serialHub *serialhub
+
+	serialPortList *SerialPortList
+
+	tools *tools.Tools
+
+	systray *systray.Systray
 }
 
-var h = hub{
-	broadcast:    make(chan []byte, 1000),
-	broadcastSys: make(chan []byte, 1000),
-	register:     make(chan *connection),
-	unregister:   make(chan *connection),
-	connections:  make(map[*connection]bool),
+func newHub(tools *tools.Tools, systray *systray.Systray) *hub {
+	broadcastSys := make(chan []byte, 1000)
+
+	onRegister := func(port *serport) {
+		broadcastSys <- []byte("{\"Cmd\":\"Open\",\"Desc\":\"Got register/open on port.\",\"Port\":\"" + port.portConf.Name + "\",\"Baud\":" + strconv.Itoa(port.portConf.Baud) + ",\"BufferType\":\"" + port.BufferType + "\"}")
+	}
+	onUnregister := func(port *serport) {
+		broadcastSys <- []byte("{\"Cmd\":\"Close\",\"Desc\":\"Got unregister/close on port.\",\"Port\":\"" + port.portConf.Name + "\",\"Baud\":" + strconv.Itoa(port.portConf.Baud) + "}")
+	}
+	serialHubub := newSerialHub(onRegister, onUnregister)
+
+	onList := func(data []byte) {
+		broadcastSys <- data
+	}
+	onErr := func(err string) {
+		broadcastSys <- []byte("{\"Error\":\"" + err + "\"}")
+	}
+	serialPortList := newSerialPortList(tools, onList, onErr)
```

Comment:
I would simplify both of this two with the `ChanWriter`
```suggestion
	serialHubub := newSerialHub(ChanWriter{Ch:broadcastSys})
	serialPortList := newSerialPortList(tools, ChanWriter{Ch:broadcastSys})
```